src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,259 @@
+            )
+
+        if parent is None:
+            parent = geometry
+        if parent is None:
+            parent = lattice
+
+        self._parent = parent
+        self._geometry = geometry
+        self._lattice = lattice
+        if self._parent is None and self._geometry is None and self._lattice is None:
+
+            def conv(k):
+                if not np.allclose(k, 0.0):
+                    warn(
+                        f"{self.__class__.__name__} cannot convert stored k-points from 1/Ang to reduced coordinates. "
+                        "Please ensure parent=Hamiltonian, geometry=Geometry, or lattice=Lattice to ensure reduced k."
+                    )
+                return k / _Bohr2Ang
+
+        else:
+
+            def conv(k):
+                return (k @ lattice.cell.T) / (2 * np.pi * _Bohr2Ang)
+
+        self._convert_k = conv
+
+    def _open(self, rewind=False):
+        """Open the file
+
+        Here we also initialize some variables to keep track of the state of the read.
+        """
+        super()._open()
+        if rewind:
+            self.seek(0)
+        # Here we initialize the variables that will keep track of the state of the read.
+        # The process for identification is done on this basis:
+        #  _ik is the current (Python) index for the k-point to be read
+        #  _state is:
+        #        -1 : the file-descriptor has just been opened (i.e. in front of header)
+        #         0 : it means that the file-descriptor is in front of k point information
+        self._state = -1
+        self._ik = 0
+
+    def _setup_parsing(self, close=True):
+        """Gets all the things needed to parse the wfsx file.
+
+        Parameters
+        -----------
+        close: bool, optional
+            Whether the file unit should be closed afterwards.
+        """
+        self._open()
+        # Read the sizes relevant to the file.
+        # We also read whether there's only gamma point information or there are multiple points
+        if self._state == -1:
+            self._sizes = self._r_next_sizes()
+            self._state = 0
+
+        if close:
+            self._close()
+
+    def _r_next_sizes(self):
+        """Determine the dimension of the data stored in the vectors file"""
+        pos = self.fh.tell()
+        # Skip past header
+        self.readline()  # Empty line
+        self.readline()  # K point
+        self.readline()  # Eigenmode index
+        self.readline()  # Frequency
+        self.readline()  # Eigenmode header
+
+        # Determine number of atoms (i.e. rows per mode)
+        natoms = 0
+        while True:
+            line = self.readline()
+            if line.startswith("Eigenmode"):
+                break
+            else:
+                natoms += 1
+        self._natoms = natoms
+
+        # Determine number of modes
+        nmodes = 1
+        while True:
+            line = self.readline()
+            if line.startswith("k            ="):
```

Comment:
I suspect that newer versions would change the spacing, could we use a `regexp` with variyng space?